### PR TITLE
codeintel: removing bloom filters

### LINKED ISSUE #34862: codeintel: Remove bloom filters
#32908 disabled the bloom filters from the query path. We haven't seen a performance degredation in two months so we can assume it's safe to delete entirely.

Remove the uses of [`DEBUG_PRECISE_CODE_INTEL_BLOOM_FILTER_BAIL_OUT`](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+DEBUG_PRECISE_CODE_INTEL_BLOOM_FILTER_BAIL_OUT&patternType=literal) so that the `!= ""` path is always taken. Simplify the resulting code and eliminate any paths that are now unreachable. The `lib/codeintel/bloomfilter` package and the `filter` column on the `lsif_references` table can also be removed.

### PR DESCRIPTION
**Pull request by Numbers88s**
This PR closes #34862 with the following:

- Removed the uses of `DEBUG_PRECISE_CODE_INTEL_BLOOM_FILTER_BAIL_OUT`
- Simplified the resulting code and eliminated any paths that were now unreachable. 
- Removed the `lib/codeintel/bloomfilter` package 
- Removed the `filter` column on the lsif_references table

## Test plan

#32908 disabled the bloom filters from the query path. We haven't seen a performance degradation in two months so we can assume it's safe to delete entirely.

### CODE REVIEW

**Review by efritz — CHANGES_REQUESTED** [review]

File: migrations/frontend/1652143849/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+name: drop_filter_column_from_lsif_references_table
```
Comment (efritz) [inline migrations/frontend/1652143849/metadata.yaml:1]: This name is out of date now

**Review by efritz — APPROVED** [review]
Main dry run passed. Two comments about things that might not need changing here.

File: internal/cmd/tracking-issue/testdata/fixtures.json [hunk 2]
```diff
@@ -2669,7 +2669,7 @@
...
...~#17960~](https://github.com/sourcegraph/sourcegraph/issues/17960); PRs: ~[#24](https://github.com/sourcegraph/lsif-test/pull/24)~)\r\n- [x] (🏁 14 days ago) escape the project root we compare paths against ([~#42~](https://github.com/sourcegraph/lsif-clang/pull/42)) :shipit:\r\n\u003c!-- END ASSIGNEE --\u003e\r\n\r\n\u003c!-- BEGIN ASSIGNEE: tjdevries --\u003e\r\n@tjdevries\r\n\r\n\r\nCompleted\r\n- [x] (🏁 18 days ago) ci: Use https:// rather than ssh for clone ([~#41~](https://github.com/sourcegraph/lsif-clang/pull/41)) :shipit:\r\n- [x] (🏁 12 days ago) cross-package type alias definitions have self-referencing semantic definition results ([~#130~](https://github.com/sourcegraph/lsif-go/issues/130); PRs: ~[#131](https://github.com/sourcegraph/lsif-go/pull/131)~) 🐛\r\n\u003c!-- END ASSIGNEE --\u003e\r\n\u003c!-- END WORK --\u003e\r\n\r\n#### Legend\r\n\r\n- 👩 Customer issue\r\n- 🐛 Bug\r\n- 🧶 Technical debt\r\n- 🎩 Quality of life\r\n- 🛠️ [Roadmap](https://docs.google.com/document/d/1cBsE9801DcBF9chZyMnxRdolqM_1c2pPyGQz15QAvYI/edit#heading=h.5nwl5fv52ess)\r\n- 🕵️ [Spike](https://en.wikipedia.org/wiki/Spike_(software_development))\r\n- 🔒 Security issue\r\n- :shipit: Pull Request\r\n",
```
[the remark this answers is not in the document]
Comment (Numbers88s) [reply internal/cmd/tracking-issue/testdata/fixtures.json:2672]: Oh it references the function `ReferenceIDsAndFilters` which change its name to `ReferenceIDs`. That's what I changed in this line. Should I leave it as `ReferenceIDsAndFilters`?

[the remark this answers is not in the document]
Comment (efritz) [reply internal/cmd/tracking-issue/testdata/fixtures.json:2672]: Yeah this is just test data for the tracking issue bot.

[the remark this answers is not in the document]
Comment (Numbers88s) [reply internal/cmd/tracking-issue/testdata/fixtures.json]: Oh gotcha, changed it back.

**Pull request merged**